docs/guides/multi-cluster.md

```diff
@@ -0,0 +1,44 @@
+# Recommended Multi-Cluster Architecture
+
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one "service mesh" that spans many clusters in one aws account, this service mesh should include the following components:
```

Comment:
* I feel like we should avoid using the term "service mesh" here, as it is not a VPC Lattice terminology. saying something like one large infrastructure will do the job.
* aws -> AWS